### PR TITLE
Implement start and shutdown function in swim.go

### LINKED ISSUE #104: [swim] implement start and shutdown function
**Detail**
implement start and shutdown function

### PR DESCRIPTION
**Pull request by junbeomlee**
resolved: #104, #95 

details:

1. Implement start function
2. Implement shutdown function
3. Add membership and member message type
4. Implement exchangeMembership function in swim.go

Test cases will be added after other functions are implemented

### CODE REVIEW

**Comment by junbeomlee** [comment]
#105

File: swim.go [hunk 1]
```diff
@@ -113,11 +113,67 @@ func messageEndpointFactory(config *Config, messageEndpointConfig MessageEndpoin
...
+	}
+
+	// Exchange membership
+	msg, err := s.messageEndpoint.SyncSend(address, pb.Message{
+		Address: s.member.Address(),
+		Id:      xid.New().String(),
+		Payload: &pb.Message_Membership{
+			Membership: membership,
+		},
+	})
```
Comment (zeroFruit) [inline swim.go:142]: I think there needs Membership message handler

Comment (zeroFruit) [reply swim.go:142]: And member who receive Membership message should sends back Membership message?

Comment (junbeomlee) [reply swim.go:142]: Yes! We need to implement membership message handle function in:
```
func (s *SWIM) handle(msg pb.Message) {

	s.handlePbk(msg.PiggyBack)

	switch msg.Payload.(type) {
	case *pb.Message_Ping:
		s.handlePing(msg)
	case *pb.Message_Ack:
		// handle ack
	case *pb.Message_IndirectPing:
		s.handleIndirectPing(msg)
	default:

	}
}
```

File: pbkstore.go [hunk 2]
```diff
@@ -38,31 +38,31 @@ const (
 // The lower the priority, the faster the queried.
 // The data has a member status, such as alive, suspected or dead.
 
-type PBkStore interface {
+type MbrStatsMsgStore interface {
 	Len() int
-	Push(pbk pb.PiggyBack)
-	Get() (pb.PiggyBack, error)
+	Push(pbk pb.MbrStatsMsg)
+	Get() (pb.MbrStatsMsg, error)
```
Comment (zeroFruit) [inline pbkstore.go]: Why changed name?

**Comment by zeroFruit** [comment]
I can't see any big problem. LGTM 👍

**Pull request merged**